toolbox/HBTReader.py

```diff
@@ -1,279 +1,686 @@
+            if len(p) > 1:
+                p = set(p)
+                p.difference_update(repo)
+                repo.update(p)
+            NewPart.append((i, list(p)))
+        comp_id = lambda x: x[0]
+        NewPart.sort(key=comp_id)
+        NewPart = [x[1] for x in NewPart]
+        return NewPart
+
+    def LoadHostHalos(self, isnap=-1, selection=None):
+        """Returns spatial properties of FoF groups for a snapshot.
+
+        Arguments:
+            isnap (int): snapshot number
+            selection (tuple): (default = None) selection query being passed on
+            to HBT file
+
+        Returns:
+            (numpy.ndarray) host haloes matching the selection
+        """
+        hosthalos = []
+        offset = 0
+        trans_index = False
+
+        if selection is None:
+            selection = np.s_[:]
+        else:
+            trans_index = isinstance(selection, numbers.Integral)
+
+        if type(selection) is list:
+            selection = tuple(selection)
+
+        with self.OpenFile(isnap, filetype="HaloSize") as hostfile:
+            nsub = hostfile["HostHalos"].shape[0]
+            if trans_index:
+                if offset + nsub > selection:
+                    try:
+                        hosthalos.append(
+                            hostfile["HostHalos"][selection - offset])
+                    except ValueError:
+                        pass
+                offset += nsub
+            else:
+                try:
+                    hosthalos.append(hostfile["HostHalos"][selection])
+                except ValueError:
+                    pass
+
+        if len(hosthalos):
+            hosthalos = np.hstack(hosthalos)
+        else:
+            hosthalos = np.array(hosthalos)
+
+        return hosthalos
+
+    def GetHostHalo(self, HostHaloId, isnap=-1):
+        """Returns spatial information of a specific FoF group.
+
+        Arguments:
+            HostHaloId (int): host halo id
+            isnap (int): snapshot number
+
+        Returns:
+            (numpy.ndarray) single FoF group in a structured array
+        """
+
+        return self.LoadHostHalos(isnap, HostHaloId)[0]
+
+    def GetSubsOfHost(self, HostHaloId, isnap=-1):
+        """Loads all subhaloes belonging to a host halo.
+
+        Uses information from ``Membership/GroupedTrackIds``.
+
+        Arguments:
+            HostHaloId (int): row number
+            isnap (int): (default = -1)
+
+        Returns:
+            (numpy.ndarray): array of ``TrackIds``, or empty array if FoF does
+                not host subhalos
+        """
+        with self.OpenFile(isnap) as subfile:
+            trackIds = subfile['Membership/GroupedTrackIds'][HostHaloId]
+
+        try:
+            result = np.hstack(
+                [self.GetSub(trackId, isnap) for trackId in trackIds])
+        except ValueError:
+            result = np.array([])
+
+        return result
+
+    def GetMergerTree(self, HostHaloId, isnap=-1, file=None):
```

Comment:
GetHostMergerTree to be clear, since this tree is based on hosthalo information and is about host haloes. Note that the subhalo hierarchy of HBT+ is already a tree of TrackIds.

It would be great if you could implement a visualization module of the trees (tree diagram) and hierarchical track structure(through tree diagram or circular treemap). If that requires uncommon library module then it would be better to implement into a separate file.